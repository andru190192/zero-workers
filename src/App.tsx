import React, { useState } from 'react';
import './App.css';
import { Country } from './components/Checkbox/types';
import CheckBox from './components/Checkbox/CheckBox';

const countries: Country[] = [
  { id: 1, name: 'India' },
  { id: 2, name: 'USA' },
  { id: 3, name: 'France' },
];

function App() {
  const [selectedCountries, setSelectedCountries] = useState<Country[]>([]);
  const [selectAll, setSelectAll] = useState<boolean>(false);

  const handleCountryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const countryName = event.target.name;
    const countryValue = event.target.value;
    const isChecked = event.target.checked;

    const updatedSelectedCountries: Country[] = isChecked
      ? [...selectedCountries, { id: +countryValue, name: countryName }]
      : selectedCountries.filter((country) => country.id !== +countryValue);
      
    setSelectedCountries(updatedSelectedCountries);
    setSelectAll(updatedSelectedCountries.length === countries.length);
  };

  const handleSelectAllChange = () => {
    const updatedSelectedCountries = selectAll ? [] : countries;
    setSelectedCountries(updatedSelectedCountries);
    setSelectAll(!selectAll);
  };

  return (
    <div>
      <label>
        <input
          type="checkbox"
          checked={selectAll}
          onChange={handleSelectAllChange}
        />
        Select All
      </label>
      <br />
      {countries.map((country) => (
        <CheckBox 
          key={country.id}
          label={country.name}
          value={country.id}
          selectedItems={selectedCountries}
          handleOnChange={handleCountryChange}
        />
      ))}
    </div>
  );
}

export default App;
