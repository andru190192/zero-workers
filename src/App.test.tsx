import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  test('renders component correctly', () => {
    const { container } = render(<App />);
    const selectAllCheckbox = screen.getByText(/Select All/i);
    expect(selectAllCheckbox).toBeInTheDocument();
    const checkboxes = screen.getAllByRole('checkbox');
    expect(checkboxes.length).toBe(4);
    expect(container).toMatchSnapshot();
  });

  test('calls handleSelectAllChange when Select All checkbox is clicked', () => {
    render(<App />);
    const selectAllCheckbox = screen.getByText(/Select All/i);
    fireEvent.click(selectAllCheckbox);
    const checkboxes = screen.getAllByRole('checkbox');
    checkboxes.forEach((checkbox) => {
      expect(checkbox).toBeChecked();
    });
    fireEvent.click(selectAllCheckbox);
    checkboxes.forEach((checkbox) => {
      expect(checkbox).not.toBeChecked();
    });
  });

  test('calls handleCountryChange when country checkbox is clicked', () => {
    render(<App />);
    const indiaCheckbox = screen.getByRole('checkbox', { name: 'India' });
    fireEvent.click(indiaCheckbox);
    const selectedCheckboxes = screen.getAllByRole('checkbox', { checked: true });
    expect(selectedCheckboxes.length).toBe(1);
    fireEvent.click(indiaCheckbox);
    const unselectedCheckboxes = screen.getAllByRole('checkbox', { checked: false });
    expect(unselectedCheckboxes.length).toBe(4);
  });
});