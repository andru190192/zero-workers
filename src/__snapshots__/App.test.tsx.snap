// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<div>
  <div>
    <label>
      <input
        type="checkbox"
      />
      Select All
    </label>
    <br />
    <div>
      <input
        id="India"
        name="India"
        type="checkbox"
        value="1"
      />
      <label
        for="India"
      >
        India
      </label>
    </div>
    <div>
      <input
        id="USA"
        name="USA"
        type="checkbox"
        value="2"
      />
      <label
        for="USA"
      >
        USA
      </label>
    </div>
    <div>
      <input
        id="France"
        name="France"
        type="checkbox"
        value="3"
      />
      <label
        for="France"
      >
        France
      </label>
    </div>
  </div>
</div>
`;
